hibernate-core/src/test/java/org/hibernate/test/hql/NullSelectExpressionTest.java

```diff
@@ -0,0 +1,179 @@
+
+/**
+ * @author Gail Badner
+ */
+@TestForIssue( jiraKey = "HHH-10757")
+public class NullSelectExpressionTest extends BaseCoreFunctionalTestCase {
+
+	@Test
+	@TestForIssue(jiraKey = "HHH-10757")
+	public void testNullSelectExpression() {
+		Session s = openSession();
+		s.getTransaction().begin();
+		Person person = new Person();
+		person.firstName = "Herman";
+		person.middleName = "Joseph";
+		person.lastName = "Munster";
+		s.persist( person );
+		s.flush();
+		s.clear();
+
+		Object result = s.createQuery(
+				"select firstName, null, lastName from NullSelectExpressionTest$Person where lastName='Munster'"
+		).uniqueResult();
+		assertTrue( Object[].class.isInstance( result ) ) ;
+		Object[] objects = (Object[]) result;
+
+		assertEquals( 3, objects.length );
+		assertEquals( "Herman", objects[0] );
+		assertNull( objects[1] );
+		assertEquals( "Munster", objects[2] );
+
+		s.getTransaction().rollback();
+		s.close();
+	}
+
+	@Test
+	@TestForIssue(jiraKey = "HHH-10757")
+	public void testAliasedNullSelectExpression() {
+		Session s = openSession();
+		s.getTransaction().begin();
+		Person person = new Person();
+		person.firstName = "Herman";
+		person.middleName = "Joseph";
+		person.lastName = "Munster";
+		s.persist( person );
+		s.flush();
+		s.clear();
+
+		Object result = s.createQuery(
+				"select firstName, null as nullValue, lastName from NullSelectExpressionTest$Person where lastName='Munster'"
+		).uniqueResult();
+		assertTrue( Object[].class.isInstance( result ) ) ;
+		Object[] objects = (Object[]) result;
+		assertEquals( 3, objects.length );
+		assertEquals( "Herman", objects[0] );
+		assertNull( objects[1] );
+		assertEquals( "Munster", objects[2] );
+
+		s.getTransaction().rollback();
+		s.close();
+	}
+
+	@Test
+	@TestForIssue(jiraKey = "HHH-10757")
+	public void testNewNullSelectExpression() {
+		Session s = openSession();
+		s.getTransaction().begin();
+		Person person = new Person();
+		person.firstName = "Herman";
+		person.middleName = "Joseph";
+		person.lastName = "Munster";
+		s.persist( person );
+		s.flush();
+		s.clear();
+
+		try {
+			Person result = (Person) s.createQuery(
+					"select new NullSelectExpressionTest$Person( id, firstName, null, lastName ) from NullSelectExpressionTest$Person where lastName='Munster'"
+			).uniqueResult();
+			fail( "should have failed with QuerySyntaxException" );
+		}
+		catch (QuerySyntaxException ex) {
+			// expected because constructor cannot be resolved;
+			// for second argument, one constructor takes a String and the other takes an Integer
+		}
+		finally {
+			s.getTransaction().rollback();
+			s.close();
+		}
+	}
+
+	@Test
+	@TestForIssue(jiraKey = "HHH-10757")
+	public void testNewCastNullSelectExpression() {
+		Session s = openSession();
+		s.getTransaction().begin();
+		Person person = new Person();
+		person.firstName = "Herman";
+		person.middleName = "Joseph";
+		person.lastName = "Munster";
+		s.persist( person );
+		s.flush();
+		s.clear();
+
```

Comment:
The following fails because of the null argument.

Using "cast" is intended to disambiguate the constructor method.